metrics/go-kit/prometheus/factory_test.go

```diff
@@ -31,13 +32,26 @@ func TestGauge(t *testing.T) {
 
 func TestHistogram(t *testing.T) {
 	f := NewFactory("namespace", "subsystem", nil)
-	hist := f.Histogram("gokit_prom_hist")
+	hist := f.Histogram("gokit.prom-hist")
 	hist.Observe(10.5)
 	m := findMetric(t, "namespace_subsystem_gokit_prom_hist")
 	require.NotNil(t, m)
 	assert.Equal(t, 10.5, m[0].GetHistogram().GetSampleSum())
 }
 
+func TestWrapper(t *testing.T) {
+	f := NewFactory("", "", nil)
+	wf := xkit.Wrap("foo", f, xkit.ScopeSeparator(":"))
```

Comment:
it's a good point. I think we should design this differently. In practice nobody care that prometheus factory is creates via xkit.Wrap, so I think a better design would be
```
metrics/
    api files
    go-kit/
       the wrapper thing
    prometheus/
       something that uses go-kit wrapper but actually returns the real metrics.Factory
```